### DISCUSSION TITLE
CUSTOM RULES

### DISCUSSION THREAD

**Discussion by rambant777** [wazuh/wazuh#27302] [Q&A]
Hi!
I want to make a rule to send notifications when changes occur in the IT-adms group. What is wrong with the regex field string, how can I change it?
```
<rule id="100086" level="10">
<if_sid>18125</if_sid>
<field name="win.system.eventID">4728|4732</field>
<regex field="win.eventdata.targetUserName">IT-adms</regex>
<description>Changes in group IT-adms</description>
<group>active_directory, group_changes</group>
```

**Comment by jftuduri** [comment]
Hello @rambant777,

The `regex` tag does not support a field attribute. You can confirm this in the Wazuh documentation: https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#regex
This is likely why your rule is not working as expected.

If you want to apply a regex to match the content of a specific field, you should use the [field](https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html#field) tag instead.

Here’s how your rule should look:

```
<rule id="100086" level="10">
	<if_sid>18125</if_sid>
	<field name="win.system.eventID">4728|4732</field>
	<field name="win.eventdata.targetUserName">IT-adms</field>
	<description>Changes in group IT-adms</description>
	<group>active_directory, group_changes</group>
</rule>
```


This rule will trigger an alert when:

- An event matches the conditions of rule `18125`.
- The `win.system.eventID` is either `4728` or `4732`.
- The `win.eventdata.targetUserName` matches `IT-adms`.

After making this change, remember to restart the Wazuh manager for it to take effect:

`systemctl restart wazuh-manager`

If it still doesn’t work as expected, please share a sample of the full event you’re trying to match, and I’ll assist you further.

Best regards!

**Reply by rambant777** [reply]
> <rule id="100086" level="10">
> 	<if_sid>18125</if_sid>
> 	<field name="win.system.eventID">4728|4732</field>
> 	<field name="win.eventdata.targetUserName">IT-adms</field>
> 	<description>Changes in group IT-adms</description>
> 	<group>active_directory, group_changes</group>
> </rule>
> ```

Hi!
I want to change this rule to send notifications for all groups with 3 letters adm, can you tell me how to do it? 
```
<field name="win.eventdatatargetUsername">*adm*</field>
```
is this correct?

**Reply by jftuduri** [reply]
If you want to match any value containing `adm`, you can use:

```
<field name="win.eventdata.targetUserName">adm</field>
```
(Note that it does not have the asterisks)
This will match values like `admin`, `user-adm`, or `adm-lab`.

On the other hand, if you want to match an exact value, such as `IT-adms`, you should use:

```
<field name="win.eventdata.targetUserName">^IT-adms$</field>
```
- `^` ensures the match starts at the beginning of the text.
- `$` ensures the match ends at the end of the text.

So this will match `IT-adms` exactly, but it will not match values like `IT-adms2` or `IT-adms-lab`.

**Reply by rambant777** [reply]
> If you want to match any value containing `adm`, you can use:
> 
> ```
> <field name="win.eventdata.targetUserName">adm</field>
> ```
> 
> (Note that it does not have the asterisks) This will match values like `admin`, `user-adm`, or `adm-lab`.
> 
> On the other hand, if you want to match an exact value, such as `IT-adms`, you should use:
> 
> ```
> <field name="win.eventdata.targetUserName">^IT-adms$</field>
> ```
> 
> * `^` ensures the match starts at the beginning of the text.
> * `$` ensures the match ends at the end of the text.
> 
> So this will match `IT-adms` exactly, but it will not match values like `IT-adms2` or `IT-adms-lab`.

Thanks!

**Comment by Damian-Mangold** [comment]
I closed it because it had been inactive for more than 7 days. Please reopen it if necessary.
